### PR TITLE
MGMT-13846: Reject UserManagedNetworking true when cluster VIPs are set

### PR DESCRIPTION
**Pull request by nmagnezi**
This change also fixes a case where in KubeAPI, the update flow did not
account for UserManagedNetworking updates.

API responses will match the logical order of invalid params.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by filanov — COMMENTED** [review]
Please add a test that check with different order, both user managed network then ip and the other way around

File: internal/cluster/validations/validations.go [hunk 1]
```diff
@@ -505,6 +505,16 @@ func ValidateClusterUpdateVIPAddresses(ipV6Supported bool, cluster *common.Clust
 
 	targetConfiguration.UserManagedNetworking = swag.Bool(false)
 	if params.UserManagedNetworking != nil {
+		if swag.BoolValue(params.UserManagedNetworking) {
+			err = ValidateVIPsWereNotSetUserManagedNetworking(cluster.APIVip, cluster.IngressVip,
+				cluster.APIVips, cluster.IngressVips, swag.BoolValue(cluster.VipDhcpAllocation))
+			if err != nil {
+				// reformat error to match order of actions
+				errParts := strings.Split(err.Error(), " cannot be set with ")
```
Comment (slaviered) [inline internal/cluster/validations/validations.go:513]: I would write the error again for the reason that on debugging we tend to search for the error message to find the failure in the code.

**Comment by eranco74** [comment]
/hold
@tsorya I have a vague memory of you using a workaround setting the cluster vips/machine cidr while using none platform.
Please take a look and unhold if it's not relevant

**Comment by tsorya** [comment]
As far as I remember currently it is blocked by installer already

File: internal/cluster/validations/validations.go (hunk 1 above)
Comment (nmagnezi) [reply internal/cluster/validations/validations.go:513]: This will duplicate a lot of code.
How does this any different from any other string formatting?

**Comment by nmagnezi** [comment]
@eranco74, @tsorya,
Given @eliorerz's response, can we `unhold` this PR?
If not, how should we proceed?

**Pull request merged**